core/src/main/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationService.scala

```diff
@@ -109,4 +112,44 @@ class GoogleProjectRegistrationService(protected val ctx: RawlsRequestContext,
         case None => Future.successful(None)
       }
     } yield finalResult
+
+  def getGoogleProjects(billingProjectName: Option[RawlsBillingProjectName]): Future[Seq[GoogleProjectRegistration]] = {
+    val accessibleGoogleProjectsFuture = samDAO
+      .listResourcesWithActions(SamResourceTypeNames.googleProject, SamGoogleProjectActions.readPolicies, ctx)
+      .map(_.map(resource => GoogleProjectId(resource.getResourceId)).toSet)
+
+    accessibleGoogleProjectsFuture.flatMap { accessibleGoogleProjects =>
+      googleProjectRegRepo.getGoogleProjectRegistrations(accessibleGoogleProjects).map { registrations =>
+        val filteredRegistrations = billingProjectName
+          .map(name => registrations.filter(_.billingProjectId == name))
+          .getOrElse(registrations)
+
+        if (billingProjectName.isDefined && filteredRegistrations.isEmpty) {
+          Seq.empty
+        } else {
+          val registeredProjectIds = filteredRegistrations.map(_.googleProjectId).toSet
+          val unregisteredProjects =
+            accessibleGoogleProjects.diff(registeredProjectIds).map(UnRegisteredGoogleProjectRegistration.apply)
```

Comment:
is it even possible to reach this case? If so, there may be a logic error somewhere. The billing-project column in the db is not nullable, so I wouldn't expect to find this at all.

I'd love to get rid of the magic "UNREGISTERED" string and the `UnRegisteredGoogleProjectRegistration` helper entirely; I think we can find more standard solutions for that